packages/ra-ui-materialui/src/input/ArrayInput/SimpleFormIterator.spec.tsx

```diff
@@ -348,6 +348,58 @@ describe('<SimpleFormIterator />', () => {
         expect(screen.queryAllByText('ra.action.remove').length).toBe(1);
     });
 
+    it('should add correct children with default value after removing one', async () => {
+        render(
+            <Wrapper>
+                <SimpleForm>
+                    <ArrayInput
+                        source="emails"
+                        defaultValue={[{ email: 'test@marmelab.com' }]}
+                    >
+                        <SimpleFormIterator>
+                            <TextInput
+                                source="email"
+                                label="CustomLabel"
+                                defaultValue="default@marmelab.com"
+                            />
+                        </SimpleFormIterator>
```

Comment:
could you also include an input with no default value?

Reply:
Done